4b/Grains/src/Test1.cs

```diff
@@ -1,14 +1,38 @@
+using System;
 using System.Threading.Tasks;
 using Interfaces;
+using Interfaces.TRX;
 
 namespace Grains
 {
     public class Test1 : Orleans.Grain, ITest1
     {
-        public async Task<bool> HelloWorldTest()
+        public async Task<UnitTest> HelloWorldTest()
         {
             await Task.Delay(1000);
-            return await Task.FromResult(true);
+
+            var methodName = Helpers.CallerName();
+            var classFullName = this.GetType().FullName;
+            var assemblyName = this.GetType().Assembly.GetName();
+            
+            var unitTest = new UnitTest()
```

Comment:
This whole construction we should probably extract into a helper so we don't repeat ourselves.